packages/dashboard/src/components/dashboard/main-menu.tsx

```diff
@@ -52,55 +88,220 @@ export const MainMenu = React.memo((props: MainMenuProps) => {
+        case 5:
+        case 0:
+          spoiltEquipment.push(lift + ` - ${liftModeToString(lifts[lift].current_mode)}`);
+          modeCounter['outOfOrder'] += 1;
+          break;
+      }
+    });
+    return {
+      item: 'Lift',
+      summary: { operational: modeCounter.operational, outOfOrder: modeCounter.outOfOrder },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getDispenserSummary = (): ItemSummary => {
+    let modeCounter = { operational: 0, outOfOrder: 0 };
+    const spoiltEquipment: string[] = [];
+    const dispensers = itemState.dispensers;
+    const dispenserKeys = Object.keys(dispensers);
+
+    dispenserKeys.forEach((dispenser) => {
+      switch (dispensers[dispenser].mode) {
+        case 0:
+        case 1:
+        case 2:
+          modeCounter['operational'] += 1;
+          break;
+        default:
+          spoiltEquipment.push(dispenser + '- Unknown');
+          modeCounter['outOfOrder'] += 1;
+          break;
+      }
+    });
+
+    return {
+      item: 'Dispensers',
+      summary: { operational: modeCounter.operational, outOfOrder: modeCounter.outOfOrder },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getRobotSummary = (): ItemSummary => {
+    let modeCounter = { operational: 0, outOfOrder: 0, charging: 0, idle: 0 };
+    const spoiltEquipment: string[] = [];
+    const fleets = itemState.robots;
+    const fleetKeys = Object.keys(fleets);
+
+    fleetKeys.forEach((fleet) => {
+      fleets[fleet].robots.forEach((robot) => {
+        switch (robot.mode.mode) {
+          case 8:
+          case 5:
+            spoiltEquipment.push(robot.name + ` - ${robotModeToString(robot.mode)}`);
+            modeCounter['outOfOrder'] += 1;
+            break;
+          case 1:
+            modeCounter['operational'] += 1;
+            modeCounter['charging'] += 1;
+            break;
+          case 7:
+          case 6:
+          case 2:
+          case 3:
+          case 4:
+            modeCounter['operational'] += 1;
+            break;
+          case 0:
+            modeCounter['operational'] += 1;
+            modeCounter['idle'] += 1;
+            break;
+        }
+      });
+    });
+    return {
+      item: 'Robots',
+      summary: {
+        operational: modeCounter.operational,
+        outOfOrder: modeCounter.outOfOrder,
+        charging: modeCounter.charging,
+        idle: modeCounter.idle,
+      },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getBannerColor = (): boolean => {
+    const getSpoiltDoor = getDoorSummary().summary.outOfOrder;
+    const getSpoiltLift = getLiftSummary().summary.outOfOrder;
+    const getSpoiltDispensers = getDispenserSummary().summary.outOfOrder;
+    const getSpoiltRobots = getRobotSummary().summary.outOfOrder;
+
+    return getSpoiltDoor + getSpoiltLift + getSpoiltDispensers + getSpoiltRobots !== 0;
+  };
+
+  const getSpoiltEquipment = (): string[] => {
+    const getSpoiltDoor = getDoorSummary().outOfOrder;
```

Comment:
nit: This and `getBannerColor` repeats the same iteration through `getDoorSummary()` etc.